File: Ywy/JAVA/src/level1/PrimeNumberFind.java
package level1;

public class PrimeNumberFind {
	public int solution(int n) {
		final boolean[] numberTable = new boolean[++n];
		int index = 2;
		int count = 2;
		do {
			if (numberTable[index] == false) {
				count += marking(numberTable, n, index);
			}
		} while(++index < n);
		return n - count;
	}
	
	private int marking(final boolean[] numberTable, final int n, final int no) {
		int temp = no;
		int count = 0;
		while((temp += no) < n) {
			if (numberTable[temp] == false) {
				count++;
				numberTable[temp] = true;
			}
		}
		return count;
	}
}

File: Ywy/JavaScript/level1/PrimeNumberFind.js
export default class PrimeNumberFind {
    solution = (n) => {
        const numberTable = {};
        let index = 2;
        let count = 2;
        n++;
        do {
            if (numberTable[index] === undefined) {
                count += this.marking(numberTable, n, index);
            }
        } while(++index < n);
        return n - count;
    };

    marking = (numberTable, n, no) => {
        let count = 0;
        let temp = no;
        while((temp += no) < n) {
            if (numberTable[temp] === undefined){
                numberTable[temp] = temp;
                count++;
            }
        }
        return count;
    }
}
